app/src/main/java/ch/epfl/polychef/fragments/RateRecipeFragment.java

```diff
@@ -0,0 +1,128 @@
+     */
+    public RateRecipeFragment() {
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_rate_recipe, container, false);
+    }
+
+    /**
+     * Called immediately after {@link #onCreateView(LayoutInflater, ViewGroup, Bundle)}
+     * as returned, but before any saved state has been restored in to the view.
+     * This gives subclasses a chance to initialize themselves once
+     * they know their view hierarchy has been completely created.  The fragment's
+     * view hierarchy is not however attached to its parent at this point.
+     * @param view The View returned by {@link #onCreateView(LayoutInflater, ViewGroup, Bundle)}.
+     * @param savedInstanceState If non-null, this fragment is being re-constructed
+     * @throws NullPointerException if this.getArguments() is null or if bundle.getSerializable("RecipeToRate") is null
+     */
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        Bundle bundle = this.getArguments();
+
+        recipe = (Recipe) bundle.getSerializable("RecipeToRate");
+
+        String text = getActivity().getString(R.string.RateText) + " \"" + recipe.getName() + "\" ?";
+        TextView rateText =  getView().findViewById(R.id.RateText);
+        rateText.setText(text);
+
+        postButton = getView().findViewById(R.id.buttonSendRate);
+        postButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                checkAndSendRate();
+            }
+        });
+
+    }
+
+    private void checkAndSendRate(){
+        Spinner spinner = getView().findViewById(R.id.RateChoices);
+        // The index returned is the same as the nb of stars
+        int starNb = spinner.getSelectedItemPosition() ;
+
+        //TODO: remove -> the index returned = nb of stars
+        /*String txt = Integer.toString(starNb);
+        */
+
+        String userID = userStorage.getPolyChefUser().getKey();
+
+        Log.w(TAG, "userID "+userID);
+
+        int oldRating = recipe.getRating().addRate(userID, starNb);
+
+        if(oldRating == -1 || oldRating==starNb) {
+            String newRatingText =  "Your rating is " + starNb +" stars.";
+            Toast.makeText(getActivity(), newRatingText , Toast.LENGTH_SHORT).show();
+        } else {
+            String newRatingText =  "Your new rating is " + starNb +" stars. Your previous rating was "+oldRating;
+            Toast.makeText(getActivity(), newRatingText , Toast.LENGTH_LONG).show();
+        }
+
+
+        DatabaseReference ref = fireDatabase.getReference(RecipeStorage.DB_NAME).child(recipe.getRecipeDatabaseKey());
+        ref.setValue(recipe);
```

Comment:
For later, we may do it in its own file for easier testing and more separation

Reply:
Je suis d'accord, on le fera plus tard dans ce cas